base/ca/src/com/netscape/ca/CAService.java

```diff
@@ -1256,73 +1342,95 @@ boolean verifySCT(CTResponse response, byte[] tbsCert, String logPublicKey)
-         * public key, calculated over the DER encoding of the key represented
-         * as SubjectPublicKeyInfo.  This is needed to bind the issuer to the
-         * final certificate.
-         */
-        X509CertImpl cacert = mCA.getCACert();
-        byte[] issuer_key = cacert.getPublicKey().getEncoded();
-        //SubjectPublicKeyInfo issuer_spki = new SubjectPublicKeyInfo(cacert.getPublicKey());
-        //byte[] issuer_key = issuer_spki.getEncoded();
-        byte[] issuer_key_hash = null;
-        try {
-            MessageDigest SHA256Digest = MessageDigest.getInstance("SHA256");
+            /** per rfc 6962 -
+             * "issuer_key_hash" is the SHA-256 hash of the certificate issuer's
+             * public key, calculated over the DER encoding of the key represented
+             * as SubjectPublicKeyInfo.  This is needed to bind the issuer to the
+             * final certificate.
+             */
+            X509CertImpl cacert = mCA.getCACert();
+            /* cfu: Todo: not sure which is expected,
+             *       (tried both but still failed at verify below):
+             *   - public key DER, or
+             *   - SubjectPublicKeyInfo DER
+             */
+            byte[] issuer_key = cacert.getPublicKey().getEncoded();
+            //SubjectPublicKeyInfo issuer_spki = new SubjectPublicKeyInfo(cacert.getPublicKey());
+            //byte[] issuer_key = issuer_spki.getEncoded();
+            byte[] issuer_key_hash = null;
 
             issuer_key_hash = SHA256Digest.digest(issuer_key);
-        } catch (NoSuchAlgorithmException ex) {
-            logger.error(method + " getting hash of CA signing key:" + ex.toString());
-            return false;
-        }
-
-        byte[] extensions = new byte[] {0, 0};
 
-        // piece them together
-
-        int data_len = version.length + signature_type.length +
-                 timestamp.length + entry_type.length +
-                 issuer_key_hash.length + tbsCert.length + extensions.length;
-
-        logger.debug(method + " data_len = "+ data_len);
-        ByteArrayOutputStream ostream = new ByteArrayOutputStream();
-
-        ostream.write(version);
-        ostream.write(signature_type);
-        ostream.write(timestamp);
-
-        ostream.write(entry_type);
-        ostream.write(issuer_key_hash);
-        ostream.write(tbsCert);
-
-        ostream.write(extensions);
-
-        byte[] data = ostream.toByteArray();
-
-        // Now, verify the signature
-        // Note: this part currently does not work; see method comment above
-
-        // cfu ToDo: interpret the alg bytes later; hardcode for now
-        Signature signer = Signature.getInstance("SHA256withEC", "Mozilla-JSS");
-        signer.initVerify(log_pubKey);
-        signer.update(data);
+            String extensions_s = response.getExtensions();
+            byte[] extensions = null;
+            int extensions_len = 0;
+            if (extensions_s != null && !extensions_s.equals("")) {
```

Comment:
nitpick: use `extensions_s.isEmpty()` instead of `!extensions_s.equals("")`

Reply:
sure. sounds good.   Will do.